### DISCUSSION TITLE
part-ii/2021-26/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#366] [Question comments]
Part II 2021 Automata and Formal Languages: Paper 4, Section I, $4 \mathrm{~F}$ 

https://questions.tripos.org/part-ii/2021-26/

**Comment by artie2000** [comment]
<details>
<summary>(i)</summary>

Yes: given by regular expression $(01)^\star01$.

</details>
<details>
<summary>(ii)</summary>

No: let $n$ be the pumping constant, and pump $0^n1^{2n}0^n$ down.

</details>
<details>
<summary>(iii)</summary>

Yes: the complement is given by the obvious DFA.

</details>
